easybuild/tools/filetools.py

```diff
@@ -273,10 +357,38 @@ def download_file(filename, url, path):
         attempt_cnt += 1
         _log.warning("Downloading failed at attempt %s, retrying..." % attempt_cnt)
 
+
     # failed to download after multiple attempts
     return None
 
 
+def download_svn(revision, url, path):
+    # import here to avoid circular import 
+    from easybuild.tools.repository.svnrepo import SvnRepository
+    # transform url into something SvnRepository understands
+    if url.startswith('svn+'):
+        url = url[4:]
+    # revision could already be a part of the url, we don't want that
+    if url.endswith(str(revision)):
+        url = url[:-len(revision)]
+    # same for path
+    if path.endswith(str(revision)):
+        path = path[:-len(revision)]
+    path = os.path.join(path, 'repo')
+    # svn revisions are digit's so only get these, some packages have version='r1234'
+    revision = ''.join([x for x in revision if x.isdigit()])
+
+    try:
+        svnrepo = SvnRepository(url)
+    except AttributeError, err:
+        _log.debug('error checkout out svn repo %s', err)
+        raise err
```

Comment:
I like the backtrace to come from the correct place, _log.error should only log, not raise exceptions imho.